src/PostList.js

```diff
@@ -20,44 +21,53 @@ class PostList extends React.Component {
     this.loadPosts()
   }
 
-  loadPosts() {
+  fetchParams() {
     let token = Cookies.get("acebookSession")
-    fetch(`${BASE_URL}/api/v1/posts`, {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json',
-        Accept: 'application/json',
-        Authorization: token
-      },
-      credentials: 'include'
+    return (
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+          Authorization: token
+        },
+        credentials: 'include'
+      }
+    )
+  }
+
+  loadPosts() {
+    this.setState({
+      awaitingResults: true
     })
+    fetch(`${BASE_URL}/api/v1/posts`, this.fetchParams())
       .then(res => res.json())
       .then(
         (result) => {
           this.setState({
-            isLoaded: true,
-            posts: result.posts
+            loadedFor: this.props.user.current.id,
+            posts: result.posts,
+            awaitingResults: false
           });
         }
       )
       .catch(err => console.log(err))
   }
 
-  setLoaded(value) {
-    this.setState({
-      isLoaded: value
-    })
-  }
-
   render() {
-    const { error, isLoaded, posts } = this.state;
+    const { error, loadedFor, posts } = this.state;
+    console.log(this.state.loadedFor);
+    console.log(this.props.user.current.id);
+    if (this.state.loadedFor !== this.props.user.current.id && !this.state.awaitingResults) {
```

Comment:
this statement is quite difficult to wrap your head around, maybe refactor for readability

Reply:
is it carried out like this: ((this.state.loadedFor !== this.props.user.current.id) && !this.state.awaitingResults)

Reply:
I have now tidied this up!